spec/models/user_spec.rb

```diff
@@ -1,63 +0,0 @@
-require 'rails_helper'
-
-RSpec.describe User, type: :model do
-  subject { User.new(name: 'Tom', photo: 'https://unsplash.com/photos', bio: 'Teacher from Mexico.') }
-  before { subject.save }
-
-  it 'name should be present' do
-    subject.name = nil
-    expect(subject).to_not be_valid
-  end
-
-  it 'bio should be present' do
-    subject.bio = nil
-    expect(subject).to_not be_valid
-  end
-
-  it 'photo should be present' do
-    subject.photo = nil
-    expect(subject).to_not be_valid
-  end
-
-  it 'name should not be short' do
-    subject.name = 'k'
-    expect(subject).to_not be_valid
-  end
-
-  it 'bio should not be short' do
-    subject.bio = 'k'
-    expect(subject).to_not be_valid
-  end
-
-  it 'post counter should be integer' do
-    subject.post_counter = 'number'
-    expect(subject).to_not be_valid
-  end
-
-  it 'post counter should be greater than zero' do
-    subject.post_counter = '-1'
-    expect(subject).to_not be_valid
-  end
-
-  it 'post counter should be equal to zero' do
-    subject.post_counter = 0
-    expect(subject).to be_valid
-  end
-
-  it 'post counter should be greather than zero' do
-    subject.post_counter = 2
-    expect(subject).to be_valid
-  end
-
-  describe 'Should test recent post method' do
-    before do
-      4.times do |post_number|
-        Post.create(author_id: subject.id, title: "this is the post #{post_number}")
-      end
-    end
-
-    it 'User should have three recent posts' do
-      expect(subject.most_recent_posts).to eq(subject.posts.last(3))
-    end
```

Comment:
- [ ] Kindly make sure to wrap all your tests in a `describe` statement, so your tests are grouped
- [ ] Also make sure to Create Unit tests for your `User` model, they got removed after you revert changes to this commit

Reply:
Okay. Thank you, I will sort it out now. I had no idea everything was removed, I thought it will undo the merged pull request only.

Reply:
Seriously I am stuck on how to rectify this problem. Do you perhaps have a better solution?

Reply:
Or a suggestion?

Reply:
@SelmaNdapanda, you could revert changes again to the last commit where the changes are not removed, or you can copy the deleted code from GitHub using the `Files changes` tab and paste everything in your local copy of the repo before pushing it again to GitHub

Reply:
Okay, Thank you. I undid the last commit using this command: git reset --hard HEAD~1
But it gave me this error when I tried to push to Github: To https://github.com/SelmaNdapanda/blog-app.git
 ! [rejected]        revert-3-validations_model_specs -> revert-3-validations_model_specs (non-fast-forward)
error: failed to push some refs to 'https://github.com/SelmaNdapanda/blog-app.git'
hint: Updates were rejected because the tip of your current branch is behind
hint: its remote counterpart. Integrate the remote changes (e.g.
hint: 'git pull ...') before pushing again.
hint: See the 'Note about fast-forwards' in 'git push --help' for details.

Reply:
The changes that I pulled are the ones with deleted files. It is Okay to recreate a new branch and a new pull request for review?

Reply:
Yes, but refer to the first pull request before this in your new PR

Reply:
@SelmaNdapanda, You can fix this by doing `git revert a453ce9`, and saving the commit then `git push origin revert-3-validations_model_specs`

Reply:
> Yes, but refer to the first pull request before this in your new PR description

I think this will do. because I have tried everything else before and none is working.  Thank you so much for your help.

Reply:
Okay. This worked. Thank you so much.

Reply:
I appreciate your help @Ismailco